src/index.js

```diff
@@ -216,21 +240,36 @@ const compare = (condition, value, attributes) => {
 const reduceRule = (rule, attributes) => {
   const result = {};
 
-  for (const [name, condition] of Object.entries(rule)) {
-    const values = getAttributeValues(attributes, name.split('.'));
+  for (let [pathToCheck, condition] of Object.entries(rule)) {
+    const { pathToCheck: newPathToCheck, condition: newCondition } =
+      maybeReverseCondition(pathToCheck, condition, attributes);
+    pathToCheck = newPathToCheck;
+    condition = newCondition;
```

Comment:
I wonder if using a `map` here might improve the readability and allow us to avoid the `let`...maybe something like this?

```suggestion
  const conditions = Object.entries(rule)
    .map(
      ([pathToCheck, condition]) =>
        maybeReverseCondition(pathToCheck, condition, attributes)
    );
      
  for (const { pathToCheck, condition } of conditions) {
    // ...
```

What do you think?

(`conditions` is not a great name)

Reply:
Hmmm - it didn't seem much more readable by glancing at it, and it did have the caveat that it iterates twice, right?

Reply:
Fair points. Resolving